pinner/pinmgr.go

```diff
@@ -297,33 +293,29 @@ func (pm *PinManager) closeQueueDataStructures() {
 	if err != nil {
 		pm.log.Fatal(err)
 	}
-
-	err = pm.duplicateGuard.Close()
-	if err != nil {
-		pm.log.Fatal(err)
-	}
 }
 
-func createLevelDBKey(value PinningOperationData, log *zap.SugaredLogger) []byte {
-	var buffer bytes.Buffer
-	enc := gob.NewEncoder(&buffer)
-	if err := enc.Encode(value.ContId); err != nil {
-		log.Fatal("Unable to encode value")
-	}
-	return buffer.Bytes()
+func createLevelDBKey(value PinningOperationData, log *zap.SugaredLogger) uint64 {
+	return value.ContId
 }
 
-func createLevelDB(QueueDataDir string, log *zap.SugaredLogger) *leveldb.DB {
-	dname := filepath.Join(QueueDataDir, "duplicateGuard")
-	err := os.MkdirAll(dname, os.ModePerm)
-	if err != nil {
-		log.Fatal("Unable to create directory for LevelDB. Out of disk? Too many open files? try ulimit -n 50000")
-	}
+func buildDupliceGuardFromPinQueue(QueueDataDir string, log *zap.SugaredLogger) map[uint64]bool {
+	ret := make(map[uint64]bool)
+	dname := filepath.Join(QueueDataDir, "pinQueueMsgPack")
 	db, err := leveldb.OpenFile(dname, nil)
 	if err != nil {
-		log.Fatal("Unable to create LevelDB. Out of disk? Too many open files? try ulimit -n 50000")
+		return ret
 	}
-	return db
+	iter := db.NewIterator(nil, nil)
+	for iter.Next() {
+		entry, err := decodeMsgPack(iter.Value())
+		if err != nil {
+			continue
+		}
+		ret[entry.ContId] = true
+	}
+	db.Close()
```

Comment:
this is throwing CVE errors as we're not checking if we successfully close the database